pkg/broker/variables.go

```diff
@@ -87,3 +92,43 @@ func (bv *BrokerVariable) ToSchema() map[string]interface{} {
 
 	return schema
 }
+
+// ValidateVariables validates a list of BrokerVariables adhere to their JSONSchema.
+func ValidateVariables(parameters map[string]interface{}, schemaVariables []BrokerVariable) error {
+
+	allErrors := &multierror.Error{
+		ErrorFormat:utils.LineErrorFormatter,
+	}
+
+	for _, variable := range schemaVariables {
+
+		value, ok := parameters[variable.FieldName]
+		if !ok {
+			// According to json schema, the required property trumps the default value.
+			if variable.Required {
+				multierror.Append(allErrors, fmt.Errorf("missing required parameter \"%s\"", variable.FieldName))
+				continue
+			} else if variable.Default != nil {
+				// Insert the default value into the parameters
+				value = variable.Default
+				parameters[variable.FieldName] = value
+			}
+		}
+
+		result, err := gojsonschema.Validate(gojsonschema.NewGoLoader(variable.ToSchema()), gojsonschema.NewGoLoader(value))
+		if err != nil {
+			multierror.Append(allErrors, err)
+		} else if len(result.Errors()) > 0 {
+			for _, r := range result.Errors() {
+				// For better output, replace the "root" keyword for the root json object to the variable name.
+				multierror.Append(allErrors, errors.New(strings.Replace(r.String(), "(root)", fmt.Sprintf("(%s)", variable.FieldName), -1)))
```

Comment:
I think it would make sense to construct a JSONSchema for the whole object and validate that. That way the schema will handle the required fields and I think you won't have to replace (root) because it should show the correct property name:

```
var required []string
properties := make(map[string]interface{})

for _, variable := range schemaVariables {
	properties[variable.FieldName] = variable.ToSchema()
	if variable.Required() {
		required = append(required, variable.FieldName)
	}
}

objectSchema := map[string]interface{}{
	"required": required,
	"properties": properties,
}

result, err := gojsonschema.Validate(gojsonschema.NewGoLoader(objectSchema), gojsonschema.NewGoLoader(parameters))
//...
```

Reply:
This is done, created a helper function `createJsonSchema`